dbr.go

```diff
@@ -13,13 +13,30 @@ import (
 // Open creates a Connection.
 // log can be nil to ignore logging.
 func Open(driver, dsn string, log EventReceiver) (*Connection, error) {
-	if log == nil {
-		log = nullReceiver
-	}
 	conn, err := sql.Open(driver, dsn)
 	if err != nil {
 		return nil, err
 	}
+
+	return OpenWith(driver, dsn, conn, log)
+}
+
+// OpenWith creates a Connection with injectable sql.DB.
+// log can be nil to ignore logging.
+// conn can be nil to use default sql impl.
+func OpenWith(driver, dsn string, conn *sql.DB, log EventReceiver) (*Connection, error) {
```

Comment:
can this be `OpenRaw(conn *sql.DB,  d Dialect, log EventReceiver)` ?

sqlmock does not specify a dialect. it is cleaner to just let the users directly pick one.

Reply:
also please add a test.

Reply:
Well, this would end in a function like this:
`func OpenRaw(conn *sql.DB, d Dialect, log EventReceiver) (*Connection, error)  {
	return &Connection{DB: conn, EventReceiver: log, Dialect: d}, nil
}`

I'm with you, that this would be the clean solution, but in this case I could simply use the Connection struct instead.